src/main/java/Duke.java

```diff
@@ -1,10 +1,92 @@
+        System.out.println("_________________________________________");
+        System.out.println("Bye. Hope to see you again soon!");
+        System.out.println("_________________________________________");
+    }
+
+    public static void printList(List<Task> list) {
+        System.out.println("_________________________________________");
+        if (list.size() > 0) {
+            System.out.println("Here are the tasks in your list:");
+            for (int i = 1; i <= list.size(); i++) {
+                System.out.println(i + "." + list.get(i - 1));
+            }
+        } else {
+            System.out.println("You have no task in the list.");
+        }
+        System.out.println("_________________________________________");
+    }
+
+    public static void addTask(Task t, List<Task> list) {
+        list.add(t);
+        System.out.println("_________________________________________");
+        System.out.println("Got it. I've added this task:");
+        System.out.println(" " + t);
+        System.out.printf("Now you have %d tasks in the list.\n", list.size());
+        System.out.println("_________________________________________");
+    }
+
+    public static void done(int index, List<Task> list) {
+        if (index > list.size() - 1) {
+            System.out.println("No such task!");
+        } else {
+            list.get(index).isDone = true;
+            System.out.println("_________________________________________");
+            System.out.println("Nice! I've marked this task as done:");
+            System.out.println(" " + list.get(index));
+            System.out.println("_________________________________________");
+        }
+    }
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
-    }
-}
+        Scanner scan = new Scanner(System.in);
+        List<Task> taskList = new ArrayList<>();
+
+        greet();
+
+        while (scan.hasNext()) {
+            String command = scan.nextLine();
+
+            if (command.equals("bye")) {
+                exit();
+                break;
+            } else if (command.equals("list")) {
+                printList(taskList);
+            } else if (command.matches("done\\s\\d*")) {
+                command = command.substring(5);
+                int index = Integer.parseInt(command) - 1;
+                done(index, taskList);
+            } else if (command.matches("todo\\s.*")) {
+                command = command.substring(5);
+                Task todo = new Todo(command);
+                addTask(todo, taskList);
+            } else if(command.matches("deadline\\s.*")) {
+                command = command.substring(9);
+                String[] arr = command.split("/by",2);
```

Comment:
I think we should try not to use arr as it may be too general, maybe something like commandWords instead?